### PR TITLE
Closes #2518 RC2 Bug Fixes introduced extra space to the main navigation

### LINKED ISSUE #2518: RC2 Bug Fixes introduced extra space to the main navigation
## Problem/Motivation
I'm noticing in new Probo builds, the primary navigation has extra spacing added above all of the navigation elements.

##  Screenshots
image

### LINKED ISSUE #2519: Determine how to SAFELY uninstall modules and remove associated permissions
## What is the problem that we want to solve?
In order to remove modules from Quickstart we need to be able to uninstall the module and remove associated permissions from our existing roles without breaking or removing our roles.

## Conditions of satisfaction
- [ ] The considered options are documented
- [ ] A recommended next step is documented
- [ ] An estimate of time and effort is provided for the recommendation
- [ ] …

### PR DESCRIPTION
**Pull request by bberndt-uaz**
## Description

This PR updates the system menu block template to wrap the entire template content in a conditional to test if the menu is empty or not. The screen reader `sr-only` class is also restored to the heading element.

## Related issues

#2518 

## How to test

Confirm that the space (the empty h2 element) above the primary nav and the utility links is removed. Verify that no other system menu blocks are negatively affected.

- [View Probo site](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2520.probo.build/)

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by trackleft** [comment]
I think we can add that sr-only class back instead of the `{% if configuration.label_display %}`

File: themes/custom/az_barrio/templates/blocks/block--system-menu-block.html.twig [hunk 1]
```diff
@@ -31,26 +31,28 @@
...
-  </h2>
-  {{ title_suffix }}
-  {# Menu. #}
-  {% block content %}
-    {{ content }}
-  {% endblock %}
-</nav>
-
+    {{ title_prefix }}
+    <h2{{ title_attributes.setAttribute('id', heading_id) }}>{{ configuration.label }}</h2>
```
Comment (bberndt-uaz) [inline themes/custom/az_barrio/templates/blocks/block--system-menu-block.html.twig:50]:
```suggestion
    <h2{{ title_attributes.setAttribute('id', heading_id).addClass('h4').addClass('mt-md-5') }}>{{ configuration.label }}</h2>
```

File: themes/custom/az_barrio/templates/blocks/block--system-menu-block.html.twig [hunk 2]
```diff
@@ -31,26 +31,28 @@
...
-  </h2>
-  {{ title_suffix }}
-  {# Menu. #}
-  {% block content %}
-    {{ content }}
-  {% endblock %}
-</nav>
-
+    {{ title_prefix }}
+    <h2{{ title_attributes.setAttribute('id', heading_id).addClass('h4').addClass('mt-md-5') }}>{{ configuration.label }}</h2>
```
Comment (bberndt-uaz) [inline themes/custom/az_barrio/templates/blocks/block--system-menu-block.html.twig:50]:
```suggestion
    <h2{{ title_attributes.setAttribute('id', heading_id) }}>{{ configuration.label }}</h2>
```

**Review by trackleft — APPROVED** [review]
Tested:
https://025d4551-37f5-46e4-ba26-0d34e8373ba1.probo.build/pages/text
https://025d4551-37f5-46e4-ba26-0d34e8373ba1.probo.build/news
https://menu-block-azs-thinktank.pantheonsite.io/our-team
https://menu-block-azs-thinktank.pantheonsite.io/person/marco-ortiz

**Comment by joeparsons** [comment]
Comparing the template files in this branch to the 2.6.x branch with whitespace changes omitted made it easier for me to review these changes.

Screen captures from https://github.com/az-digital/az_quickstart/compare/2.6.x...issue/2518?w=1:

Screenshot 2023-06-23 at 3 54 24 PM

Screenshot 2023-06-23 at 3 54 40 PM

**Review by joeparsons — APPROVED** [review]

**Pull request merged**